File: src/python/associative_array.py
# Creating an associative array using a dictionary
associative_array = {}

associative_array["apple"] = 3
associative_array["banana"] = 5
associative_array["orange"] = 2

print(f"apple: {associative_array['apple']}")
print(f"banana: {associative_array['banana']}")
print(f"orange: {associative_array['orange']}")

for key, value in associative_array.items():
    print(f"{key}: {value}")

File: src/cpp/associative_array.cpp
#include <iostream>
#include <map>
#include <string>

int main() {
  std::map<std::string, int> associativeArray;

  associativeArray["apple"] = 3;
  associativeArray["banana"] = 5;
  associativeArray["orange"] = 2;

  std::cout << "apple: " << associativeArray["apple"] << std::endl;
  std::cout << "banana: " << associativeArray["banana"] << std::endl;
  std::cout << "orange: " << associativeArray["orange"] << std::endl;

  for (const auto& pair : associativeArray) {
    std::cout << pair.first << ": " << pair.second << std::endl;
  }

  return 0;
}
